python-checks/src/main/java/org/sonar/python/checks/DjangoRenderContextCheck.java

```diff
@@ -78,33 +82,56 @@ private static void checkIfLocalsIsAssignedToContextParameter(SubscriptionContex
         List<Tree> assignmentStmts = contextSymbol.usages().stream()
           .filter(usage -> usage.kind() == Usage.Kind.ASSIGNMENT_LHS)
           .map(Usage::tree)
-          .map(usage -> TreeUtils.firstAncestorOfKind(usage, Tree.Kind.ASSIGNMENT_STMT))
+          .map(usage -> TreeUtils.firstAncestorOfKind(usage, Tree.Kind.ASSIGNMENT_EXPRESSION, Tree.Kind.ASSIGNMENT_STMT, Tree.Kind.ANNOTATED_ASSIGNMENT))
+          .filter(Objects::nonNull)
           .collect(Collectors.toList());
         if (assignmentStmts.size() == 1) {
-          AssignmentStatement assignment = (AssignmentStatement) assignmentStmts.get(0);
-          checkIfLocalsIsCalledOrFindTheNextAncestor(ctx, contextArg, assignment, recursionCount);
+          Tree assignment = assignmentStmts.get(0);
+          
+          Expression assignedValue = null;
+          if(assignment.is(Tree.Kind.ASSIGNMENT_STMT)){
```

Comment:
make auto-formatting of the class, otherwise it will be marked as change once someone will call the formatting in the future